statefun-sdk-js/src/message.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as types from "./types";
+import {validateTypeName, Address, isEmptyOrNull, Type} from "./core";
+
+class Message {
+    readonly #targetAddress;
+    readonly #typedValue: any;
+
+    constructor(targetAddress: Address, typedValue: any) {
+        this.#targetAddress = targetAddress;
+        this.#typedValue = typedValue;
+    }
+
+    get targetAddress() {
+        return this.#targetAddress;
+    }
+
+    get valueTypeName() {
```

Comment:
So far this was "typename" everywhere, not "typeName". It'd be nice to make this consistent, and my vote would be changing it to typeName everywhere.